dev-tools/mage/clean.go

```diff
@@ -0,0 +1,49 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"github.com/magefile/mage/sh"
+)
+
+// CleanPaths specifies a list of files or paths to recursively delete.
+// The values may contain variables and will be expanded at the time of use.
+var CleanPaths = []string{
+	"build",
+	"docker-compose.yml.lock",
+	"{{.BeatName}}",
+	"{{.BeatName}}.exe",
+	"{{.BeatName}}.test",
+	"{{.BeatName}}.test.exe",
+	"fields.yml",
+	"_meta/fields.generated.yml",
+	"_meta/kibana.generated",
+	"_meta/kibana/5/index-pattern/{{.BeatName}}.json",
+	"_meta/kibana/6/index-pattern/{{.BeatName}}.json",
+}
+
+// Clean clean generated build artifacts.
+func Clean() error {
```

Comment:
There are a many defaults/magic paths here. How about making it somewhat configurable by passing the paths as arguments?

```
func Clean(pathsList ...[]string) error {
  if len(pathsList) == 0 {
    pathsList = [][]string{CleanPaths}
  }
  for _, lst := range pathsList {
    for _, f := range lst {
      ..
    }
  }
}
```

So the per beat magefile can be written like:

```
func Clean() error {
  return make.Clean(mage.DefaultCleanPaths, []string{
    "...",
    "...",
  })
}
```

This is still valid.

```
func Clean() error {
  return mage.Clean()
}
```

Reply:
Changed. 👍